### ISSUE TITLE
Protobuf names that have an _ (underscore) cause errors

### ISSUE THREAD

**Issue by lewissk** [trueaccord/ScalaPB#170]
Using version 0.4.20

If your protobuf looks like this (test project is located here https://github.com/lewissk/test_protobuf/tree/master):

```protobuff
syntax = "proto2";
package test;

option java_package = "test";
option cc_enable_arenas = true;

enum ContentCategory {
  IAB1_1 = 2;
  IAB11 = 191;
}
```

Then you run:

  `sbt compile`

You will get the following error:

    [error] /Users/xxx/test/protobuf_scala_test/target/src_managed/main/compiled_protobuf/test/ContentCategory.scala:11: method isIab11 is defined twice
    [error]   conflicting symbols both originated in file '/Users/xxx/protobuf_scala_test/target/src_managed/main/compiled_protobuf/test/ContentCategory.scala'
    [error]   def isIab11: Boolean = false
    [error]       ^
    [error] one error found
    [error] (compile:compileIncremental) Compilation failed

Because the scala it generates is like so:

````scala
// Generated by the Scala Plugin for the Protocol Buffer Compiler.
// Do not edit!

package test


import com.trueaccord.scalapb.Descriptors

sealed trait ContentCategory extends com.trueaccord.scalapb.GeneratedEnum {
  def isIab11: Boolean = false
  def isIab11: Boolean = false
}

object ContentCategory extends com.trueaccord.scalapb.GeneratedEnumCompanion[ContentCategory] {
  @SerialVersionUID(0L)
  case object IAB1_1 extends ContentCategory {
    val id = 2
    val name = "IAB1_1"
    override def isIab11: Boolean = true
  }

  @SerialVersionUID(0L)
  case object IAB11 extends ContentCategory {
    val id = 191
    val name = "IAB11"
    override def isIab11: Boolean = true
  }

  lazy val values = Seq(IAB1_1, IAB11)
  def fromValue(value: Int): ContentCategory = value match {
    case 2 => IAB1_1
    case 191 => IAB11
  }
  lazy val descriptor = new Descriptors.EnumDescriptor(0, "ContentCategory", this)
}
```

Other language compilers that I have tried don't remove the _ (underscore) they leave them in.

Java generates:
```java
// Generated by the protocol buffer compiler.  DO NOT EDIT!
// source: test.proto

package test;

public final class Test {
  private Test() {}
  public static void registerAllExtensions(
      com.google.protobuf.ExtensionRegistry registry) {
  }
  /**
   * Protobuf enum {@code test.ContentCategory}
   */
  public enum ContentCategory
      implements com.google.protobuf.ProtocolMessageEnum {
    /**
     * <code>IAB1_1 = 2;</code>
     */
    IAB1_1(0, 2),
    /**
     * <code>IAB11 = 191;</code>
     */
    IAB11(1, 191),
    ;

    /**
     * <code>IAB1_1 = 2;</code>
     */
    public static final int IAB1_1_VALUE = 2;
    /**
     * <code>IAB11 = 191;</code>
     */
    public static final int IAB11_VALUE = 191;


    public final int getNumber() { return value; }

    public static ContentCategory valueOf(int value) {
      switch (value) {
        case 2: return IAB1_1;
        case 191: return IAB11;
        default: return null;
      }
    }

    public static com.google.protobuf.Internal.EnumLiteMap<ContentCategory>
        internalGetValueMap() {
      return internalValueMap;
    }
    private static com.google.protobuf.Internal.EnumLiteMap<ContentCategory>
        internalValueMap =
          new com.google.protobuf.Internal.EnumLiteMap<ContentCategory>() {
            public ContentCategory findValueByNumber(int number) {
              return ContentCategory.valueOf(number);
            }
          };

    public final com.google.protobuf.Descriptors.EnumValueDescriptor
        getValueDescriptor() {
      return getDescriptor().getValues().get(index);
    }
    public final com.google.protobuf.Descriptors.EnumDescriptor
        getDescriptorForType() {
      return getDescriptor();
    }
    public static final com.google.protobuf.Descriptors.EnumDescriptor
        getDescriptor() {
      return test.Test.getDescriptor().getEnumTypes().get(0);
    }

    private static final ContentCategory[] VALUES = values();

    public static ContentCategory valueOf(
        com.google.protobuf.Descriptors.EnumValueDescriptor desc) {
      if (desc.getType() != getDescriptor()) {
        throw new java.lang.IllegalArgumentException(
          "EnumValueDescriptor is not for this type.");
      }
      return VALUES[desc.getIndex()];
    }

    private final int index;
    private final int value;

    private ContentCategory(int index, int value) {
      this.index = index;
      this.value = value;
    }

    // @@protoc_insertion_point(enum_scope:test.ContentCategory)
  }


  public static com.google.protobuf.Descriptors.FileDescriptor
      getDescriptor() {
    return descriptor;
  }
  private static com.google.protobuf.Descriptors.FileDescriptor
      descriptor;
  static {
    java.lang.String[] descriptorData = {
      "\n\ntest.proto\022\004test*)\n\017ContentCategory\022\n\n" +
      "\006IAB1_1\020\002\022\n\n\005IAB11\020\277\001B\006\n\004test"
    };
    com.google.protobuf.Descriptors.FileDescriptor.InternalDescriptorAssigner assigner =
        new com.google.protobuf.Descriptors.FileDescriptor.    InternalDescriptorAssigner() {
          public com.google.protobuf.ExtensionRegistry assignDescriptors(
              com.google.protobuf.Descriptors.FileDescriptor root) {
            descriptor = root;
            return null;
          }
        };
    com.google.protobuf.Descriptors.FileDescriptor
      .internalBuildGeneratedFileFrom(descriptorData,
        new com.google.protobuf.Descriptors.FileDescriptor[] {
        }, assigner);
  }

  // @@protoc_insertion_point(outer_class_scope)
}
```

C++ generates
```c++
// Generated by the protocol buffer compiler.  DO NOT EDIT!
// source: test.proto

#ifndef PROTOBUF_test_2eproto__INCLUDED
#define PROTOBUF_test_2eproto__INCLUDED

#include <string>

#include <google/protobuf/stubs/common.h>

#if GOOGLE_PROTOBUF_VERSION < 2006000
#error This file was generated by a newer version of protoc which is
#error incompatible with your Protocol Buffer headers.  Please update
#error your headers.
#endif
#if 2006001 < GOOGLE_PROTOBUF_MIN_PROTOC_VERSION
#error This file was generated by an older version of protoc which is
#error incompatible with your Protocol Buffer headers.  Please
#error regenerate this file with a newer version of protoc.
#endif

#include <google/protobuf/generated_message_util.h>
#include <google/protobuf/repeated_field.h>
#include <google/protobuf/extension_set.h>
#include <google/protobuf/generated_enum_reflection.h>
// @@protoc_insertion_point(includes)

namespace test {

// Internal implementation detail -- do not call these.
void  protobuf_AddDesc_test_2eproto();
void protobuf_AssignDesc_test_2eproto();
void protobuf_ShutdownFile_test_2eproto();


enum ContentCategory {
  IAB1_1 = 2,
  IAB11 = 191
};
bool ContentCategory_IsValid(int value);
const ContentCategory ContentCategory_MIN = IAB1_1;
const ContentCategory ContentCategory_MAX = IAB11;
const int ContentCategory_ARRAYSIZE = ContentCategory_MAX + 1;

const ::google::protobuf::EnumDescriptor* ContentCategory_descriptor();
inline const ::std::string& ContentCategory_Name(ContentCategory value) {
  return ::google::protobuf::internal::NameOfEnum(
    ContentCategory_descriptor(), value);
}
inline bool ContentCategory_Parse(
    const ::std::string& name, ContentCategory* value) {
  return ::google::protobuf::internal::ParseNamedEnum<ContentCategory>(
    ContentCategory_descriptor(), name, value);
}
// ===================================================================


// ===================================================================


// ===================================================================


// @@protoc_insertion_point(namespace_scope)

}  // namespace test

#ifndef SWIG
namespace google {
namespace protobuf {

template <> struct is_proto_enum< ::test::ContentCategory> : ::google::protobuf::internal::true_type {};
template <>
inline const EnumDescriptor* GetEnumDescriptor< ::test::ContentCategory>() {
  return ::test::ContentCategory_descriptor();
}

}  // namespace google
}  // namespace protobuf
#endif  // SWIG

// @@protoc_insertion_point(global_scope)

#endif  // PROTOBUF_test_2eproto__INCLUDED
```

Python generates

```python
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: test.proto

import sys
_b=sys.version_info[0]<3 and (lambda x:x) or (lambda x:x.encode('latin1'))
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
from google.protobuf import descriptor_pb2
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()




DESCRIPTOR = _descriptor.FileDescriptor(
  name='test.proto',
  package='test',
  serialized_pb=_b('\n\ntest.proto\x12\x04test*)\n\x0f\x43ontentCategory\x12\n\n\x06IAB1_1\x10\x02\x12\n\n\x05IAB11\x10\xbf\x01\x42\x06\n\x04test')
)
_sym_db.RegisterFileDescriptor(DESCRIPTOR)

_CONTENTCATEGORY = _descriptor.EnumDescriptor(
  name='ContentCategory',
  full_name='test.ContentCategory',
  filename=None,
  file=DESCRIPTOR,
  values=[
    _descriptor.EnumValueDescriptor(
      name='IAB1_1', index=0, number=2,
      options=None,
      type=None),
    _descriptor.EnumValueDescriptor(
      name='IAB11', index=1, number=191,
      options=None,
      type=None),
  ],
  containing_type=None,
  options=None,
  serialized_start=20,
  serialized_end=61,
)
_sym_db.RegisterEnumDescriptor(_CONTENTCATEGORY)

ContentCategory = enum_type_wrapper.EnumTypeWrapper(_CONTENTCATEGORY)
IAB1_1 = 2
IAB11 = 191


DESCRIPTOR.enum_types_by_name['ContentCategory'] = _CONTENTCATEGORY


DESCRIPTOR.has_options = True
DESCRIPTOR._options = _descriptor._ParseOptions(descriptor_pb2.FileOptions(), _b('\n\004test'))
# @@protoc_insertion_point(module_scope)

```

**Comment by thesamet** [comment]
What version are you using?

**Comment by lewissk** [comment]
I am using version 0.4.20

**Comment by thesamet** [comment]
I believe this has been fixed in 0.5.x. can you try?

**Comment by lewissk** [comment]
I should have dug further into the versioning. It is fixed in 0.5.40
